### PR TITLE
Modular parsing and code migration

### PR DESCRIPTION
**Pull request by mlr0p**
This PR implements modular parsing to reduce code reuse, and also migrates the old code to the new template #11 .

### CODE REVIEW

File: importers/util/common.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+	}
+	return string(runes)
+}
+
+// MakeImporter constructs and returns an importer
+func MakeImporter(defaultFileName string, parser LineParser, numThreads int) (*Importer, error) {
+
+	// Get command-line flags
+	verbose := flag.Bool("v", false, "Displays progress bar")
+	fileNamePtr := flag.String("i", defaultFileName, "Path to dump file")
```
Comment (SoftPoison) [inline importers/util/common.go]: as you're dereferencing the fileNamePtr anyway, might as well just change this to `fileName := *flag.String(...)`

File: importers/util/custom-parse.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ParseSQL parses a line of the SQL statement by extracting
+// the relevant INSERT line into the target table
+func ParseSQL(line string, table string) ([]string, error) {
```
Comment (SoftPoison) [inline importers/util/custom-parse.go:10]: This should be fine for now, but I'm not sure if this is the most resilient parser

File: importers/forbes.go [hunk 3]
```diff
@@ -1,17 +1,19 @@
 package main
 
+/***
```
Comment (SoftPoison) [inline importers/forbes.go]: Is this comment meant to be here?

File: importers/util/common.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
+// Package util provides common code for all the importers
```
Comment (SoftPoison) [inline importers/util/common.go:1]: Since we're using GenericData for almost everything, perhaps we move the struct in here?

**Review by SoftPoison — CHANGES_REQUESTED** [review]
Good work overall, couple of small comments here and there but nothing major :)

File: importers/forbes.go (hunk 3 above)
Comment (mlr0p) [reply importers/forbes.go]: oh no...

File: importers/util/custom-parse.go (hunk 2 above)
Comment (mlr0p) [reply importers/util/custom-parse.go:10]: will keep an eye out for this in the future then, all others are fixed, hopefully

**Review by SoftPoison — APPROVED** [review]
LGTM

**Review by ss23 — APPROVED** [review]
LGTM 👍

**Pull request merged**